workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/request/RequestExecutor.java

```
@@ -1,49 +1,62 @@
+import java.util.List;
 import java.util.Map;
 
 @Slf4j
+@Component
 public class RequestExecutor implements ActivityExecutor<ExecuteRequest> {
 
   private static final String OUTPUT_STATUS_KEY = "status";
   private static final String OUTPUT_BODY_KEY = "body";
 
+  private final HttpClient httpClient;
+
+  public RequestExecutor(HttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
   @Override
   public void execute(ActivityExecutorContext<ExecuteRequest> execution) throws JsonProcessingException {
     ExecuteRequest activity = execution.getActivity();
-    Object data;
-    int statusCode;
 
     try {
-      ApiResponse<Object> apiResponse = execution.bdk()
-          .apiClient(activity.getUrl())
-          .invokeAPI("", activity.getMethod(), Collections.emptyList(), activity.getBody(), activity.getHeaders(),
-              Collections.emptyMap(), Collections.emptyMap(), "application/json", "application/json", null,
-              new TypeReference<>() {});
-
-      data = apiResponse.getData();
-      statusCode = apiResponse.getStatusCode();
-    } catch (ApiException apiException) {
-      statusCode = apiException.getCode();
-
-      // we need to set the responseBody as a map instead of a string to allow processing it in subsequent activities
-      data = new ObjectMapper().readValue(apiException.getResponseBody(), Map.class);
-      log.debug("This error happens when the request fails.", apiException);
+      Response response =
+          this.httpClient.execute(activity.getMethod(), activity.getUrl(), activity.getBody(),
+              headersToString(activity.getHeaders()));
+
+      execution.setOutputVariable(OUTPUT_STATUS_KEY, response.getCode());
+      execution.setOutputVariable(OUTPUT_BODY_KEY, response.getContent());
+    } catch (IOException ioException) {
+      log.debug("This error happens when the request fails with IOException.", ioException);
+      throw new RuntimeException(ioException.getMessage());
```

Exceptions should be wrapped so we have a proper stacktraces, here we are loosing the original exception (new RunTimeException(ioException))

---

beside we already throw ioexception from the executors, do we need to catch it then? (for all BDK calls right now we let the ioexception be raised)